OHHTTPStubs/Sources/NSURLSession/NSMutableURLRequest+HTTPBodyTesting.m

```diff
@@ -0,0 +1,89 @@
+/***********************************************************************************
+*
+* Copyright (c) 2016 Sebastian Hagedorn
+*
+* Permission is hereby granted, free of charge, to any person obtaining a copy
+* of this software and associated documentation files (the "Software"), to deal
+* in the Software without restriction, including without limitation the rights
+* to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+* copies of the Software, and to permit persons to whom the Software is
+* furnished to do so, subject to the following conditions:
+*
+* The above copyright notice and this permission notice shall be included in
+* all copies or substantial portions of the Software.
+*
+* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+* IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+* FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+* AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+* LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+* OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+* THE SOFTWARE.
+*
+***********************************************************************************/
+
+#import "NSMutableURLRequest+HTTPBodyTesting.h"
+#import <objc/runtime.h>
+
+NSString *const OHHTTPStubs_HTTPBodyKey = @"OHHTTPStubs_HTTPBodyKey";
+
+@implementation NSURLRequest (CustomHTTPBodyProperty)
+
+- (NSData *)OHHTTPStubs_HTTPBody {
+    return [NSURLProtocol propertyForKey:OHHTTPStubs_HTTPBodyKey inRequest:self];
+}
+
+@end
+
+@implementation NSMutableURLRequest (TestSwizzling)
+
+/*
+ *   Courtesy of http://nshipster.com/method-swizzling/
+ */
+
++ (void)load {
+    static dispatch_once_t onceToken;
+
+    dispatch_once(&onceToken, ^{
+        Class class = [self class];
+
+        SEL originalSelector = @selector(setHTTPBody:);
```

Comment:
I agree that the static variable is easier to grasp than the commented message sending approach. I'll take a look how we can refactor this to reuse the swizzling code.